nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UnpackContent.java

```diff
@@ -309,7 +309,10 @@ public void process(final InputStream in) throws IOException {
                             }
                             final File file = new File(tarEntry.getName());
                             final Path filePath = file.toPath();
-                            final String filePathString = filePath.getParent() + "/";
+                            String filePathString = "/";
+                            if(filePath.getParent() != null) {
+                                filePathString = filePath.getParent() + "/";
```

Comment:
The `/` should be moved to the front of the string concatenation for consistency and so that we don't have trailing separators. Once you do that, we can merge.

Reply:
Do you mean replacing e.g. "ham/egg/" with "/ham/egg"? Because that would change prior behavior of the "file.path" attribute. In that case, this behavior has to be changed in some other places too (e.g. all the other unpackers).

Reply:
Ok, let's skip that then.